Sources/Shared/Core/BlueprintLayout.swift

```diff
@@ -253,6 +253,13 @@
   /// - Parameter indexPath: The index path of the item whose attributes are requested.
   /// - Returns: A layout attributes object containing the information to apply to the item’s cell.
   override open func layoutAttributesForItem(at indexPath: IndexPath) -> LayoutAttributes? {
+    guard cachedItems.count > 0 else {
+        return nil
+    }
+    guard indexPath.section < cachedItems.count else {
+        return nil
+    }
```

Comment:
Would it be nicer if we combine the guard statements?

```swift
guard cachedItems.count > 0 && indexPath.section < cachedItems.count else {
  return nil
}
```

What do you think?

Reply:
Perhaps this could be a method an internal method on `BlueprintLayout` ?

```swift
func indexPathIsInBounds(_ indexPath: IndexPath,for cache: [[LayoutAttributes]]) -> Bool {
  return cachedItems.count > 0 && indexPath.section < cachedItems.count
}
```

Then we could use the same logic at all call-sites.

Reply:
@zenangst - makes perfect sense